### DISCUSSION TITLE
Human initialization issue

### DISCUSSION THREAD

**Discussion by nicoeiris11** [vladmandic/human#341] [Q&A] [answered]
Hi Vlad!

I'm experiencing a weird behavior when initializing the human library in my electron js app running in the browser.
(human version: 3.0.5, OSx 11.4 M1 Mac Mini, Google Chrome Version 109.0.5414.119)

1. Even though I explicit `backend: webgl` in my human config, while debugging human initialization I can see that the library is trying to set `tensorflow` as the backend (throwing errors). 
2. In addition, after instantiating human, I can see that `human.env` has `node: true` and `browser: true` when node field should be disabled, right? (I checked this env object in the human browser demo and `node: false`, `browser: true`)

Here are some screenshots of this situation: 
- Human config running on the electron app
- Human initialization debugging logs
- Human env object

![image](https://user-images.githubusercontent.com/17889735/221224781-adfdc53a-cb6d-42a2-8e76-2d41641d3bce.png)

![image](https://user-images.githubusercontent.com/17889735/221224958-8e1846dc-cdeb-4905-8641-b57bc8d3087a.png)

![image](https://user-images.githubusercontent.com/17889735/221225407-a39b9294-4a50-44ee-b2a1-a768a5f6a26e.png)

Any suggestion or idea about this? I'm afraid `node: true` negatively affects the performance of the pipeline.

**Comment by vladmandic** [comment, marked as answer]
`electron` provides both nodejs and browser contexts, so it really depends where and how you're loading the library.
but basically, you'll need to find an `electron` setting to disable node context for a page where you're loading `human`.

i can probably add some additional intelligence to handle this as `electron` is a common use-case. but then again, i cannot know if user wants to run in browser or node context? there are valid use-cases for both.

**Reply by nicoeiris11** [reply]
Thank you Vlad! I'll check it out.

Do you think having both node and browser enabled in human env negatively affects performance?

**Reply by vladmandic** [reply]
it doesn't impact, it just generates strange warnings during backend selection and makes it impossible to select backend manually correctly.
but at the end, only one backend is actually active.

**Comment by vladmandic** [comment]
i've just added some extra detection code to better handle situations like this, new version is on github  
(its not on npmjs as its too minor of a patch to force a version bump)
